src/nsidc/metgen/config.py

```diff
@@ -31,6 +31,9 @@ class Config:
     number: int
     dry_run: bool
 
+    def __post_init__(self):
+        self.collection = None
```

Comment:
@eigenbeam The most obvious approach (to me) for splitting out the Collection object is to manage it with the other configuration. However, it's derived from configuration input and it doesn't feel right to set up the Collection at config initialization. Let me know if you have any thoughts from an architectural point of view..